src/Spd.Resource.Repository/PortalUser/PortalUserRepository.cs

```diff
@@ -35,7 +35,11 @@ public async Task<PortalUserListResp> QueryAsync(PortalUserQry qry, Cancellation
             users = users.Where(d => d.spd_servicecategory == null || d.spd_servicecategory == (int)PortalUserServiceCategoryOptionSet.Screening);
         else
             users = users.Where(d => d.spd_servicecategory == (int)PortalUserServiceCategoryOptionSet.Licensing);
-
+        if (qry.ContactRoleCode != null)
+        {
+            Guid categoryKey = DynamicsContextLookupHelpers.RoleGuidDictionary.FirstOrDefault(x => x.Key == qry.ContactRoleCode.ToString()).Value;
+            users = users.Where(u => u.spd_spd_role_spd_portaluser.Any(r => r.spd_roleid == categoryKey));

```

Comment:
this probably won't work and we need to query 2 role code. 
Suggest you just use spd_servicecategory

Reply:
Fixed.